.github/workflows/unit_tests.yml

```diff
@@ -0,0 +1,70 @@
+name: Unit tests
+
+on:
+  pull_request:
+    branches:
+      - main
+
+concurrency:
+  group: ${{ github.ref }}-${{ github.workflow }}
+  cancel-in-progress: true
+
+env:
+  CARGO_TERM_COLOR: always
+  CARGO_INCREMENTAL: 0
+
+jobs:
+  checks:
+    name: Run clippy and unit tests
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v3
+
+      - name: Run cargo fmt
+        uses: actions-rs/cargo@v1
+        with:
+          command: fmt
+          args: --all
+
+      - name: Run clippy
```

Comment:
Clippy needs to download dependencies (similarly the "run unit tests" job) so we should recover cache before these steps as well.

Reply:
It takes ~1 minute to do this whole job without caching, sounds like a bit of an overkill

Reply:
Ok, agreed.